process/eventsInterceptor.go

```diff
@@ -0,0 +1,96 @@
+package process
+
+import (
+	"encoding/hex"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/notifier-go/data"
+)
+
+// ArgsEventsInterceptor defines the arguments needed for cresating an events interceptor instance
+type ArgsEventsInterceptor struct {
+	PubKeyConverter core.PubkeyConverter
+}
+
+type eventsInterceptor struct {
+	pubKeyConverter core.PubkeyConverter
+}
+
+// NewEventsInterceptor created a new eventsInterceptor instance
+func NewEventsInterceptor(args ArgsEventsInterceptor) (*eventsInterceptor, error) {
+	if check.IfNil(args.PubKeyConverter) {
+		return nil, ErrNilPubKeyConverter
+	}
+
+	return &eventsInterceptor{
+		pubKeyConverter: args.PubKeyConverter,
+	}, nil
+}
+
+// ProcessBlockEvents will process block events data
+func (ei *eventsInterceptor) ProcessBlockEvents(eventsData *data.ArgsSaveBlockData) *data.SaveBlockData {
+	events := ei.getLogEventsFromTransactionsPool(eventsData.TransactionsPool.Logs)
```

Comment:
this code can panic here as eventsData is a pointer. Should first check if eventsData is nil, then eventsData.TransactionsPool should be checked as well since it is a pointer too.

question: why TransactionsPool is a pointer? in order to avoid copying a huge structure?

Reply:
yes, added nil checks and error handling here